File: src/items/Backstage.py
from src.Items.NormalItem import *

'''
This class will create objects of Backstage type
'''

class Backstage(NormalItem):

    def __init__(self, name, sell_in, quality):

        Item.__init__(self, name, sell_in, quality)

    def updateQuality(self):
        
        if 0 < self.sell_in < 5:

            self.setQuality()
        
        elif self.sell_in :

            self.setQuality()
        
        elif self.sell_in :

            self.setQuality()

File: java/domain/Backstage.java
package java.domain;

public class Backstage extends NormalItem {
    
    public Backstage(String name, int sellIn, int quality){

        super(name, sellIn, quality);
    }

    @Override
    public void updateQuality(){

        if (this.getSellIn()>10){
            this.calculateQuality(+1);

        }

        else if (this.getSellIn()>5){
            this.calculateQuality(+2);
        }

        else if (this.getSellIn() > 0){
            this.calculateQuality(3);
        }

        else {

            this.setQuality(0);
        }
        this.setSellIn();
    }
}
